lazy_dataset/core.py

```diff
@@ -453,6 +453,48 @@ def map(self, map_fn: callable, num_workers: int = 0,
             )
         return MapDataset(map_fn, self)
 
+    def batch_map(self, map_fn: callable, num_workers: int = 0,
+                  buffer_size: int = 100, backend: str = 't'):
+        """Applies map_fn to each element in each batch of the dataset.
+        Requires dataset to be batched, i.e., the items of the
+        dataset have to be iterable.
+
+        Args:
+            map_fn:
+            num_workers:
+            buffer_size:
+            backend:
+
+        Returns:
+
+        >>> import lazy_dataset
+        >>> ds = lazy_dataset.new({'a': 1, 'b': 2, 'c': 3}, name='MyDataset')
+        >>> ds
+          DictDataset(name='MyDataset', len=3)
+        MapDataset(_pickle.loads)
+        >>> def foo(x): return 2*x
+        >>> ds1 = ds.map(foo)
```

Comment:
How about using `ds1 = ds.map(foo).batch(1)`?
Then both (`ds1` and `ds2`) would yield the same examples?